### PR TITLE
Introduce a ringbuffer

### PR DESCRIPTION
**Pull request by steveklabnik**
This is the first step towards handling keyboards properly.

### CODE REVIEW

File: src/keyboard/src/lib.rs [hunk 1]
```diff
@@ -17,6 +21,8 @@ pub static STATE: Mutex<Modifiers> = Mutex::new(Modifiers {
 	caps: false,
 });
 
+static mut BUFFER: RingBuffer<u8> = RingBuffer::new();
```
Comment (steveklabnik) [inline src/keyboard/src/lib.rs]: really, Keyboard should probably be static here...

**Comment by steveklabnik** [comment]
Also, I used an Option to try and keep things safe; this can be done with unsafe code as well, and it's more efficient, since we have an extra byte for the tag. But I think that's fine.

**Comment by nxnfufunezn** [comment]
Nice idea to have ring buffer here..

**Comment by chrisccerami** [comment]
I had never heard of a ring buffer before reading this PR, so I'm not super knowledgable, but having read up on it, this seems 👍

**Comment by steveklabnik** [comment]
I might come back to this at some point, but this PR isn't gonna get merged. Closing!

**Pull request closed without merging**